### DISCUSSION TITLE
RollUps failing for Minimum Date (fields) when where clause is present

### DISCUSSION THREAD

**Discussion by jayeshtejwani** [jamessimone/apex-rollup#302] [Q&A] [answered]
Hi @jamessimone,

Found one more issue related to Rollup, but creating discussion, I could be wrong.

I have created Custom Metadata to calculate Minimum Date on Child Records with specific criteria on Child records. When I run full recalc it fails with the Null-pointer exception. Below is the screenshot of the Debug Logs.

![Screen Shot 2022-04-20 at 2 35 29 PM](https://user-images.githubusercontent.com/18471403/164301092-71c01d27-47a3-49c9-b12b-99ff259b476e.png)

If you see the highlighted section: SuperReturnVal is being assigned null and thats the reason for throwing exception.

Also, strange thing is the minimum date after calculation should not be the default value which is being displayed in the log for the custom roll-up field Video_Flight_start_date__c 

`{
  "attributes" : {
    "type" : "Opportunity",
    "url" : "/services/data/v54.0/sobjects/Opportunity/0068C000003FjfOQAS"
  },
  "Video_Flight_start_date__c" : "1970-01-01",
  "Id" : "0068C000003FjfOQAS",
  "CurrencyIsoCode" : "USD"
}`

Adding the screenshot for the Custom metadata to help you further

![Screen Shot 2022-04-20 at 2 53 21 PM](https://user-images.githubusercontent.com/18471403/164302359-243a8063-09ea-4c64-b936-a48a635964db.png)

**Important Note: 

1. IF I remove the Where clause, calculations work perfectly fine to calculate min date value.**
2. Tested MAX(Date) with where clause, they dont work and they are failing.
3. Tested Count(num of records) operation with where clause, the result count was 0(Zero) despite of records present with filter conditions. Assuming something is not working with where clause.

**Comment by jamessimone** [comment, marked as answer]
@jayeshtejwani #304 will fix the NPE exception you've found, thanks for that. Let me know if there's anything else I can do to help on your third bullet point.

**Comment by jayeshtejwani** [comment]
@jamessimone  Thank you for acknowledging the issue. NPE is fixed but min date was calculated, end result was NULL.

**Reply by jamessimone** [reply]
@jayeshtejwani I would have to see a log with the details to dig in further, if there's an issue with the calc item where clause I would expect no matches to be found.

**Reply by jayeshtejwani** [reply]
@jamessimone 

Here are some snippets from the log. The records are queried correctly filtering the ones with where condition. But somewhere the calculation is going wrong. First test I tried keeping the parent roll-up field as null then I kept some random value, in both cases the end result on roll-up field was null.

**Lookup record prior**
![Screen Shot 2022-04-21 at 8 57 20 AM](https://user-images.githubusercontent.com/18471403/164464741-d01a017f-92b1-44ed-a3ec-9c40cbccc83f.png)

**Lookup record after**
![Screen Shot 2022-04-21 at 8 57 31 AM](https://user-images.githubusercontent.com/18471403/164464744-6cd64ff5-0716-48b2-95f6-93b626deb969.png)

**Calculated Items**
![Screen Shot 2022-04-21 at 9 01 13 AM](https://user-images.githubusercontent.com/18471403/164464748-aa4aea05-0340-4bf5-9ed7-f7bf598fb726.png)

**Reply by jayeshtejwani** [reply]
**Adding screenshot for Rollup Custom metadata as well**

![Screen Shot 2022-04-21 at 9 12 44 AM](https://user-images.githubusercontent.com/18471403/164465561-7f1b4fa7-ce71-42cc-a723-58788a22abf1.png)

**Comment by jamessimone** [comment]
You are saying `AOT_Guaranteed_Check__c = 'Video` but the value on your calc items is actually `VIDEO`. Please update your where clause to contain the right casing.

**Reply by jayeshtejwani** [reply]
@jamessimone Thank you for that input, it helped. Sorry for the inconvenience. It would be good to have a note that value need to be case-sensitive.

Since SOQL is case insensitive so I was assuming it would work because I did see query returning correct records in debug logs but as I think during comparison it would be failing.

**Reply by jamessimone** [reply]
@jayeshtejwani it's been a _long_ time since I've familiarized myself with whether or not SOQL was case-insensitive, so thanks for the reminder. I'll get a release published to remove the case sensitivity from the `RollupEvaluator`
